test/tests/functional/pbs_reservations.py

```diff
@@ -883,3 +883,128 @@ def test_fail_confirm_resv_message(self):
         # The scheduler should log reason why it was denied
         self.scheduler.log_match(rid + ";PBS Failed to confirm resv: " +
                                  "Insufficient amount of resource: ncpus")
+
+    def common_steps(self):
+        """
+        This function has common steps for configuration used in tests
+        """
+        a = {'resources_available.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+        self.server.manager(MGR_CMD_SET, SERVER, {
+            'job_history_enable': 'True'}, id=self.server.shortname)
+
+    @skipOnCpuSet
+    def test_advance_reservation_with_job_array(self):
+        """
+        Test to submit a job array within a advance reservation
+        Check if the reservation gets confimed and the jobs
+        inside the reservation starts running when the reservation runs.
+        """
+        self.common_steps()
+        # Submit a job-array
+        j = Job(TEST_USER)
+        j.set_sleep_time(10)
+        j.set_attributes({ATTR_J: '1-4:1'})
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'B'}, jid)
+        self.server.expect(JOB, {'job_state=R': 4}, count=True,
+                           id=jid, extend='t')
+        # Check status of the sub-job using qstat -fx once job completes
+        self.server.expect(JOB, {'job_state': 'F'}, extend='x',
+                           offset=10, id=jid)
+
+        # Submit a advance reservation (R1) and an array job to the reservation
+        # once reservation confirmed
+        r = Reservation(TEST_USER)
+        now = int(time.time())
+        a = {'reserve_start': now + 20,
+             'reserve_end': now + 120}
+        r.set_attributes(a)
```

Comment:
Consider passing a into Reservation() instead of calling set_attributes() (and for the other reservation)